### PR TITLE
Gemm self & accuracy check feature

### PR DESCRIPTION
**Pull request by jkottiku**
1. Validate gemm output for consistency/repeatability by self-check (current vs previous output).
2. Validate gemm output for accuracy by accuracy-check (device (GPU) output vs host (CPU) output).

### CODE REVIEW

File: gst.so/src/gst_worker.cpp [hunk 1]
```diff
@@ -489,6 +498,38 @@ bool GSTWorker::do_gst_stress_test(int *error, std::string *err_description) {
...
+    if (self_check || accu_check) {
+
+      if (error_inject) {
+        gpu_blas->set_gemm_error(error_freq, error_count);
+      }
+
+      double self_error = 0.0;
+      double accu_error = 0.0;
+
+      gpu_blas->validate_gemm(self_check, accu_check, self_error, accu_error);
```
Comment (manoj-freyr) [inline gst.so/src/gst_worker.cpp:510]: unlikely case , but we must ensure atleast 2 runs of this call to compare current and previous results.  how to ensure this ?

File: include/rvs_blas.h [hunk 2]
```diff
@@ -219,6 +237,12 @@ class rvs_blas {
     void release_host_matrix_mem(void);
     float fast_pseudo_rand(uint64_t *nextr, size_t i);
 
+    template <typename T>
+      bool check_result_consistency(void * dout, uint64_t size, double &error);
```
Comment (manoj-freyr) [inline include/rvs_blas.h:241]: size_t is always a better type for this than uint

File: gst.so/src/gst_worker.cpp (hunk 1 above)
Comment (jkottiku) [reply gst.so/src/gst_worker.cpp:510]: In src/rvs_blas.cpp, function check_result_consistency() check for comment /* Exit first iteration of self-check as there is no previous result yet ! */ its done here.

**Review by manoj-freyr — APPROVED** [review]
Update decsriptions

**Pull request merged**